app/src/main/java/com/infomaniak/mail/data/models/Mailbox.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Infomaniak kMail - Android
+ * Copyright (C) 2022 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.data.models
+
+import android.util.Log
+import com.google.gson.annotations.SerializedName
+import com.infomaniak.mail.data.api.ApiRepository
+import com.infomaniak.mail.data.cache.MailRealm
+import com.infomaniak.mail.data.cache.MailboxContentController
+import com.infomaniak.mail.utils.AccountUtils
+import io.realm.MutableRealm.UpdatePolicy
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+import io.realm.toRealmList
+
+class Mailbox : RealmObject {
+    var uuid: String = ""
```

Comment:
Isn't it a primary key?

Reply:
No, this UUID is unfortunately not different for each User.
If several Users share the same Mailbox, they'll get the same UUID.

That's why we have an unique `objectId` as primary key.